tests/typescript/conformance/classes/constructorDeclarations/constructorParameters/__snapshots__/jsfmt.spec.js.snap

```diff
@@ -0,0 +1,432 @@
+
+class C {
+    constructor(readonly x: number) {}
+}~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+// @declaration: true
+
+class C {
+  constructor(readonly x: number) {}
+}
+
+`;
+
+exports[`readonlyConstructorAssignment.ts 1`] = `
+// Tests that readonly parameter properties behave like regular readonly properties
+
+class A {
+    constructor(readonly x: number) {
+        this.x = 0;
+    }
+}
+
+class B extends A {
+    constructor(x: number) {
+        super(x);
+        // Fails, x is readonly
+        this.x = 1;
+    }
+}
+
+class C extends A {
+    // This is the usual behavior of readonly properties:
+    // if one is redeclared in a base class, then it can be assigned to.
+    constructor(readonly x: number) {
+        super(x);
+        this.x = 1;
+    }
+}
+
+class D {
+    constructor(private readonly x: number) {
+        this.x = 0;
+    }
+}
+
+// Fails, can't redeclare readonly property
+class E extends D {
+    constructor(readonly x: number) {
+        super(x);
+        this.x = 1;
+    }
+}
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+// Tests that readonly parameter properties behave like regular readonly properties
+
+class A {
+  constructor(readonly x: number) {
+    this.x = 0;
+  }
+}
+
+class B extends A {
+  constructor(x: number) {
+    super(x);
+    // Fails, x is readonly
+    this.x = 1;
+  }
+}
+
+class C extends A {
+  // This is the usual behavior of readonly properties:
+  // if one is redeclared in a base class, then it can be assigned to.
+  constructor(readonly x: number) {
+    super(x);
+    this.x = 1;
+  }
+}
+
+class D {
+  constructor(private readonly x: number) {
+    this.x = 0;
+  }
+}
+
+// Fails, can't redeclare readonly property
+class E extends D {
+  constructor(readonly x: number) {
+    super(x);
+    this.x = 1;
+  }
+}
+
+`;
+
+exports[`readonlyInAmbientClass.ts 1`] = `
+declare class C{
+	constructor(readonly x: number);
+	method(readonly x: number);
+}~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+class C {
+  constructor(readonly x: number)
+  method(x: number)
+}
+
+`;
+
+exports[`readonlyInConstructorParameters.ts 1`] = `
+class C {
+    constructor(readonly x: number) {}
+}
+new C(1).x = 2;
+
+class E {
+    constructor(readonly public x: number) {}
+}
+
+class F {
+    constructor(private readonly x: number) {}
+}
+new F(1).x;~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+class C {
+  constructor(readonly x: number) {}
+}
+new C(1).x = 2;
+
+class E {
+  constructor(public readonly x: number) {}
+}
+
+class F {
+  constructor(private readonly x: number) {}
+}
+new F(1).x;
+
+`;
+
+exports[`readonlyReadonly.ts 1`] = `
+class C {
+    readonly readonly x: number;
+    constructor(readonly readonly y: number) {}
+}~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+class C {
+  x: number;
```

Comment:
The AST just has a boolean `isReadonly`, so this seems like expected output. A number of the TypeScript tests contain intentional syntax errors, this looks like one of them. (I'm actually surprised the parser didn't throw on this one).